### PR TITLE
feat(site): add annotation to display values of type clibase.Duration correctly

### PR DESCRIPTION
**Pull request by johnstcn**
- Adds an annotation `format_duration_ns` to all deployment values of type `clibase.Duration`
- Adds a unit test that complains if you forget to add the above annotation to a `clibase.Duration`
- Modifies `optionValue()` to check for the presence of `format_duration_ns` when displaying an option.

### CODE REVIEW

File: site/src/components/DeploySettingsLayout/optionValue.test.ts [hunk 1]
```diff
@@ -37,14 +39,16 @@ describe("optionValue", () => {
         ...defaultOption,
         name: "Session Duration",
         value: 3600 * 1e9,
+        annotations: { format_duration_ns: "falsae" },
```
Comment (johnstcn) [inline site/src/components/DeploySettingsLayout/optionValue.test.ts:42]: self-review: technically a typo but illustrates behaviour well

File: codersdk/deployment.go [hunk 2]
```diff
@@ -1502,10 +1508,11 @@ when required by your organization's security policy.`,
 			// The default value is essentially "forever", so just use 100 years.
 			// We have to add in the 25 leap days for the frontend to show the
 			// "100 years" correctly.
-			Default: ((100 * 365 * time.Hour * 24) + (25 * time.Hour * 24)).String(),
-			Value:   &c.MaxTokenLifetime,
-			Group:   &deploymentGroupNetworkingHTTP,
-			YAML:    "maxTokenLifetime",
+			Default:     ((100 * 365 * time.Hour * 24) + (25 * time.Hour * 24)).String(),
```
Comment (johnstcn) [inline codersdk/deployment.go:1511]: Note: this value currently does not display correctly (screenshot from dev.coder.com):

image

We can modify this in a separate PR but I would consider it out of scope here.

**Review by mtojek — APPROVED** [review]
LGTM, but what is the story with `_ns`? Can't we use `_ms` or will it be breaking?

**Comment by johnstcn** [comment]
> LGTM, but what is the story with `_ns`? Can't we use `_ms` or will it be breaking?

It doesn't really matter what the annotation is called. I chose the `_ns` suffix because it accepts an integer value that is a number of nanoseconds.

**Comment by johnstcn** [comment]
> > LGTM, but what is the story with `_ns`? Can't we use `_ms` or will it be breaking?
> 
> It doesn't really matter what the annotation is called. I chose the `_ns` suffix because it accepts an integer value that is a number of nanoseconds.

Updated to `format_duration` without the `_ns` suffix.

**Pull request merged**